lib/config-generator.js

```diff
@@ -0,0 +1,253 @@
+/**
+ * @fileoverview to generate config files.
+ * @author 唯然<weiran.zsd@outlook.com>
+ */
+import process from "process";
+import path from "path";
+import { spawnSync } from "child_process";
+import { writeFile } from "fs/promises";
+import enquirer from "enquirer";
+import { isPackageTypeModule, installSyncSaveDev, checkPackageJson, fetchPeerDependencies } from "./utils/npm-utils.js";
+import * as log from "./utils/logging.js";
+
+/**
+ * Class representing a ConfigGenerator.
+ */
+export class ConfigGenerator {
+
+    /**
+     * Create a ConfigGenerator.
+     * @param {Object} options The options for the ConfigGenerator.
+     * @param {string} options.cwd The current working directory.
+     * @param {Object} options.answers The answers provided by the user.
+     * @returns {ConfigGenerator} The ConfigGenerator instance.
+     */
+    constructor(options = {}) {
+        this.cwd = options.cwd || process.cwd();
+        this.answers = options.answers || {};
+        this.result = {
+            devDependencies: ["eslint"],
+            configFilename: "eslint.config.js",
+            configContent: ""
+        };
+
+        const packageJsonExists = checkPackageJson(this.cwd);
+
+        if (!packageJsonExists) {
+            throw new Error("A package.json file is necessary to initialize ESLint. Run `npm init` to create a package.json file and try again.");
+        }
```

Comment:
Checking for `package.json` is a side effect in this constructor that wouldn't be expected. Can we move this to `prompt()`?

I'm also wondering if maybe instead of throwing and error, what if we just run `npm init -y` in this case?

Reply:
> Checking for package.json is a side effect in this constructor that wouldn't be expected. Can we move this to prompt()?

It's in the constructor as I think it's better to throw the error earlier. and when passed `--config`, the `prompt()` won't be called.

> I'm also wondering if maybe instead of throwing and error, what if we just run npm init -y in this case?

I prefer throwing an error here - it's possible that the user is in the wrong cwd, then running `npm init -y` does not make sense.

Reply:
The problem with the current setup is that we make people answer a bunch of questions and then throw an error, which is really frustrating. 

What if instead of throwing an error, we tell the user that a `package.json` file is required and ask if they'd like us to run `npm init -y` for them?

Reply:
order of execution: `constructor()` => `prompt()` => `output()`.

So what you're saying doesn't seem to be happening. 🤔

Reply:
Ah okay, I think I got confused trying to follow this through.

I'd still prefer that this not be in the constructor. Just instantiating a class shouldn't cause us to check for `package.json`

Reply:
it's to check if the passed `opts.cwd` is valid. While I agree that throwing errors in the constructor needs to be done cautiously, it seems appropriate in this case.

Another way is to move it to the top of `bin/create-config.js`: https://github.com/eslint/create-config/blob/issue51/bin/create-config.js. thoughts?